### PR TITLE
Allow comma separated values (List) for Layout Renderers

### PR DESCRIPTION
**Pull request by 304NotModified**
Support voor comma separated list,

e.g.

supported:
- Generic List (`List<T>`), with the following types: 
  - C# built in types (string, int, double, object)
  - enum (use shortname)
  - culture, encoding, Type
  - not supported: Layout

Not supported:
-  Arrays
- Non-generic `List` 
-  `IList<T>` / `IList`
- `IEnumerable<T>` / `IEnumerable` 

Example:

```
${layoutrenderer-with-list:enums=Ignore, Neutral, ignore}
```

``` c#
[LayoutRenderer("layoutrenderer-with-list")]
public class LayoutRendererWithListParam : LayoutRenderer
{
    public List<FilterResult> Enums { get; set; }
   protected override void Append(StringBuilder builder, LogEventInfo logEvent)
   {
       ...
   }
}
```

TO DO
- [x] escape comma -> backtick
-  [x] replace escape with optional quoted
- [x] add unit test for multiple enum

---

This change is [Reviewable](https://reviewable.io/reviews/nlog/nlog/1439)

### CODE REVIEW

File: src/NLog/Internal/PropertyHelper.cs [hunk 1]
```diff
@@ -221,15 +225,15 @@ private static bool TryNLogSpecificConversion(Type propertyType, string value, o
...
+        private static bool TryGetEnumValue(Type resultType, string value, out object result, bool flagsEnumAllowed)
         {
             if (!resultType.IsEnum)
             {
                 result = null;
                 return false;
             }
 
-            if (resultType.IsDefined(typeof(FlagsAttribute), false))
+            if (flagsEnumAllowed && resultType.IsDefined(typeof(FlagsAttribute), false))
```
Comment (bhaeussermann) [inline src/NLog/Internal/PropertyHelper.cs]: What if it's a flags-enum, but `flagsEnumAllowed` is false? The else-part would then execute with the flags-enum.

Comment (bhaeussermann) [inline src/NLog/Internal/PropertyHelper.cs]: How about adding a test for parsing a list of values of a flag-enum?

**Comment by bhaeussermann** [comment]
Any idea why the test coverage dropped so dramatically? I don't seem to be able to view the codecov diff.

File: src/NLog/Internal/PropertyHelper.cs (hunk 1 above)
Comment (304NotModified) [inline src/NLog/Internal/PropertyHelper.cs]: Good idea

**Comment by 304NotModified** [comment]
> Any idea why the test coverage dropped so dramatically? I don't seem to be able to view the codecov diff.

It did not. 75.30% (+<.01%) compared to 717373a

but this PR has "only"  74.29% coverage

**Comment by 304NotModified** [comment]
I will remove the ecape and will move to a optional quoted approach, 

eg. `a,b,c` or `'a,b',c`

**Comment by 304NotModified** [comment]
docs: https://github.com/NLog/NLog/wiki/Properties-constraints-for-custom-extensions

**Pull request merged**